NebulaPatcher/Patches/Dynamic/GalacticTransport_Patch.cs

```diff
@@ -0,0 +1,127 @@
+﻿using HarmonyLib;
+using NebulaWorld;
+using NebulaModel.Packets.Logistics;
+using UnityEngine;
+
+namespace NebulaPatcher.Patches.Dynamic
+{
+    [HarmonyPatch(typeof(GalacticTransport))]
+    class GalacticTransport_Patch
+    {
+        [HarmonyPostfix]
+        [HarmonyPatch("SetForNewGame")]
+        public static void SetForNewGame_Postfix()
+        {
+            if(SimulatedWorld.Initialized && !LocalPlayer.IsMasterClient)
+            {
+                LocalPlayer.SendPacket(new ILSRequestgStationPoolSync());
+            }
+        }
+
+        [HarmonyPrefix]
+        [HarmonyPatch("AddStationComponent")]
+        public static bool AddStationComponent_Prefix(GalacticTransport __instance, int planetId, StationComponent station)
+        {
+            if (!SimulatedWorld.Initialized || LocalPlayer.IsMasterClient)
+            {
+                return true;
+            }
+            // if we are a client and have a fake station already saved, update it with the full data given now
+            // this should happen when this client arrives at a planet for the first time whichs interplanetar logistic
+            // he was already syncing
+            while (__instance.stationRecycleCursor > 0 && __instance.stationPool[__instance.stationRecycle[__instance.stationRecycleCursor - 1]] != null)
+            {
+                Debug.Log("increase because " + (__instance.stationRecycleCursor - 1) + " is ptr to " + __instance.stationRecycle[__instance.stationRecycleCursor - 1] + " which is " + GameMain.galaxy.PlanetById(__instance.stationPool[__instance.stationRecycle[__instance.stationRecycleCursor - 1]].planetId).displayName);
+                __instance.stationRecycleCursor++;
+                if(__instance.stationRecycleCursor >= __instance.stationPool.Length)
+                {
+                    __instance.stationRecycleCursor = 0;
+                }
+            }
+            int debug = 0;
+            if (__instance.stationRecycleCursor > 0)
+            {
+                Debug.Log((__instance.stationPool[__instance.stationRecycle[__instance.stationRecycleCursor - 1]] == null) ? "ITS NULL" : "ITS NOT NULL");
+                Debug.Log("recycle " + __instance.stationRecycleCursor);
+                debug = __instance.stationRecycle[__instance.stationRecycleCursor-1];
```

Comment:
Uhm, you shouldn't do -1 here. `stationRecycle[0]` is _never_ used. Only indices [1..capacity]

Reply:
![decrement](https://user-images.githubusercontent.com/79335616/115391086-59204c80-a1df-11eb-9101-385ff2fb2675.jpg)
it does access index 0 too :)